### ISSUE TITLE
Page shifts right on context-menu 

### ISSUE THREAD

**Issue by volkandkaya** [palantir/blueprint#375]
### Bug report

- __Package version(s)__: 1.3.1
- __Browser and OS versions__:54.0.2840.100 Ubuntu 14.04

#### Steps to reproduce

1. Go to http://blueprintjs.com/docs/#components.context-menu
2. Right click and the page shifts

#### Actual behavior

Moves the page to the right

#### Expected behavior

Page doesn't move

**Comment by llorca** [comment]
I cannot repo on my end. Anyone on Chrome 54 can confirm? No problem on Chrome 55.

**Comment by giladgray** [comment]
I also can't reproduce on Chrome 54. We're both on MacOS though, with no access to Ubuntu boxes for testing.

**Comment by giladgray** [comment]
@volkandkaya could you please try a hard refresh (ctrl+shift+r) and see if that resolves the issue? wondering if your browser is serving you cached old code instead of fetching the new stuff?

**Comment by volkandkaya** [comment]
@giladgray I also checked on Opera and Firefox. Must be my Laptop, but very weird that it happens on all browsers. Hopefully someone with Ubuntu can test it.

**Comment by DaryaVeretennikova** [comment]
Page shifts because the scrollbar takes some width and the right click added class 'pt-overlay-open' to body tag with overflow: hidden.

**Comment by cmslewis** [comment]
It's possible to repro this on Mac by making the following change in System Preferences > General:

![image](https://cloud.githubusercontent.com/assets/443450/21465892/dcbd055c-c97a-11e6-86fb-75b3b3d20c4f.png)

Here's the bug in action:

![context-menu](https://cloud.githubusercontent.com/assets/443450/21465897/0cccb56c-c97b-11e6-84b1-bc7d3aab7f6d.gif)

The trickiness here is that from an interaction perspective, native OS X context menus actually block scrolling in the same way that our Blueprint `<ContextMenu>`s currently do, but they do so while keeping the scroll bar visible. There might be some sneaky way to replicate this behavior with our custom context menus, but nothing I've found in my initial searching seems particularly foolproof or elegant.

Here's a GIF of native context menus blocking scrolling, by the way:

![context-menu-native](https://cloud.githubusercontent.com/assets/443450/21465938/65ae94ba-c97c-11e6-88d2-01a16097d1db.gif)

**Comment by mayhewj** [comment]
I noticed this as well while browsing the documentation. Context Menus and Alerts both cause the scrollbar to disappear while they're active, causing a small but noticeable shift. It's happening in Chrome and Firefox on Linux.

**Comment by giladgray** [comment]
so the root cause of the disappearing scrollbar is quite simple: we're setting `overflow: hidden` on `body.pt-overlay-open`. removing that property fixes the disappearing scrollbar, but of course allows the user to scroll the page. so we're back to square one.

since `body` is the scroll container, there's very little we can do to actually prevent page scrolling manually (dang that one line of CSS was elegant).

we could use `Overlay` `backdropProps` to cancel mousewheel events which trivially prevents mouse scrolling but does nothing for keyboard scrolling. canceling keyboard scrolling is trickier because there are a lot of ways to do it (arrow keys, pgUp/Down, home/end). we could ignore those keys but there are still ways to make the browser scroll, such as a simple cmd+f for a word at the bottom. i don't think there's anything we can do about that.

so in short, i have no idea how to solve this.

**Comment by adidahiya** [comment]
overlays have changed a lot in the last year. comment if you can repro this in 2.0 and want this issue re-opened.

**Comment by dandv** [comment]
I still saw this problem on Chrome 70 on Ubuntu on the [context demo page](https://blueprintjs.com/docs/#core/components/context-menu), which is why I came to this issue. The page would shift briefly while the scrollbar was created. I've tried to reproduce it but it appears to depend on a particular navigation sequence through the demos at blueprintjs.com.
